internal/guest/runtime/runc/container.go

```diff
@@ -389,6 +400,70 @@ func (c *container) startProcess(
 		}
 	}
 
+	// This is to enabling container logging in Bare Metal Containers. This is in experimental stage now and need to be revisited.
+	var stdoutFifoPipe, stderrFifoPipe *os.File
+	if annotations != nil {
+		pipeNameSuffix, exists := annotations["io.microsoft.bmc.logging.pipelocation"]
+		pipeDirectory := "/run/gcs/containerlogs/"
+		stdoutPipeName := pipeDirectory + pipeNameSuffix + "-stdout"
+		stderrPipeName := pipeDirectory + pipeNameSuffix + "-stderr"
+		os.MkdirAll(pipeDirectory, os.ModePerm)
+		if exists {
+			_, err = os.Stat(stdoutPipeName)
+			if err != nil {
+				// fifo pipe does not exist, create one
+				err = syscall.Mkfifo(stdoutPipeName, 0666)
+				if err != nil {
+					msg := "Error creating named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+					logrus.Infof(msg)
+					return nil, errors.Wrapf(err, msg)
+				}
+			}
+
+			_, err = os.Stat(stderrPipeName)
+			if err != nil {
+				// fifo pipe does not exist, create one
+				err = syscall.Mkfifo(stderrPipeName, 0666)
+				if err != nil {
+					msg := "Error creating named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+					logrus.Infof(msg)
+					return nil, errors.Wrapf(err, msg)
+				}
+			}
+
+			// pipe either exist before hand or we have created one above
+			stdoutFifoPipe, err = os.OpenFile(stdoutPipeName, os.O_RDWR|os.O_APPEND, os.ModeNamedPipe)
+			if err != nil {
+				msg := "Error opening named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+				logrus.Infof(msg)
+				return nil, errors.Wrapf(err, msg)
+			}
+
+			stderrFifoPipe, err = os.OpenFile(stderrPipeName, os.O_RDWR|os.O_APPEND, os.ModeNamedPipe)
+			if err != nil {
+				msg := "Error opening named pipe:" + fmt.Errorf("outer error context: %w", err).Error()
+				logrus.Infof(msg)
+				return nil, errors.Wrapf(err, msg)
+			}
+		}
+
+		isLoggingSideCarContainerStr, exists := annotations["io.microsoft.bmc.logging.isLoggingSideCarContainer"]
+		if exists {
+			isLoggingSideCarContainer, _ := strconv.ParseBool(isLoggingSideCarContainerStr)
+			if !isLoggingSideCarContainer {
+				// workload container needs to redirect stdout and stderr to fifo pipe.
+				cmd.Stdout = stdoutFifoPipe
+				cmd.Stderr = stderrFifoPipe
```

Comment:
Why are we setting these here and not just in the above section where we create the pipes?

Reply:
these need to be set only for workload container and not for logging side car container.. i.e. when isLoggingSideCarContainer annotation is false. when that annotation is true, we need to specify the FDs

Reply:
Does that mean that we expect to set "io.microsoft.bmc.logging.pipelocation" on the logging container too? Why would we we do that?